yellow-paper/docs/circuits/private-kernel-reset.md

```diff
@@ -0,0 +1,160 @@
+# Private Kernel Circuit - Reset
+
+:::info Disclaimer
+This is a draft. These requirements need to be considered by the wider team, and might change significantly before a mainnet release.
+:::
+
+## Requirements
+
+The **reset** circuit is designed to abstain from processing individual private function calls. Instead, it injects the outcomes of an initial or inner private kernel circuit, scrutinizes the public inputs, and resets the verified data within its scope. This circuit can be executed either preceding the tail circuit or as a means to "reset" public inputs, allowing data to accumulate seamlessly in subsequent iterations.
```

Comment:
should "verified data" be "verifiable data"? Either way, it would be helpful to define what this data is in the `Requirements` section